torch/_inductor/codegen/cpp_micro_gemm.py

```diff
@@ -510,15 +510,16 @@ class CppMicroGemmAMX(CppMicroGemm):
 {{declare_kernel}} {
     {{kernel.assert_function}}(N % {{block_n}} == 0, "N dimension must be multiple of {{block_n}}");
     {{kernel.assert_function}}(K % 2 == 0, "K dimension must be multiple of 2");
-{%- if input_dtype == torch.bfloat16 and input2_dtype == torch.int8 %}
+{%- if use_cached_dequantized_B %}
     // Create a stack-allocated buffer for tiles of B.
-    // Except maybe for the tail-case, an AMX tile of B has 16x32 BF16 elements
+    // Except maybe for the tail-case, an AMX tile of B has 16x32 BF16 elements,
+    // but to make things simple, we'll allocate space for a 16x32 tile even for the tail.
     // It's possible that N may not be equal to Nr, but might be a multiple of it.
     // In that case, Kc (which is K in the microkernel) would be smaller (than the case in which N == Nr).
     const auto num_elements_per_b_tile = 512;
     const auto buf_size_per_nr_block = ((K + {{block_k}} - 1) / {{block_k}}) * num_elements_per_b_tile * 2;
     const auto buf_size = buf_size_per_nr_block * (N / {{block_n}});
-    alignas(4096) {{input_t}} bf16_weights_buf[buf_size];
+    alignas(4096) {{input_t}} dequantized_B_buf[buf_size];
```

Comment:
Let's use heap allocation only for MSVC compiler then. Please add a comment to note the assumption that `K` has to be small enough to guarantee the stack allocation can work well.